core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -123,12 +123,12 @@ case class LogReadResult(info: FetchDataInfo,
     this.info.records,
     this.divergingEpoch,
     this.lastStableOffset,
-    this.info.abortedTransactions,
+    if (this.info.abortedTransactions.isPresent) Some(this.info.abortedTransactions.get().asScala.toList) else None,
```

Comment:
The last `toList` results in a copy of the collection, we'd want to avoid that.

Reply:
Agree.